src/foam/foobar/FoobarTemplateUtil.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```

Comment:
Could FOAM's current templates work for this?

Reply:
Tried that and almost pulled all my hair out. Can't remember why it didn't work. This grammar is mostly copied from the template parser.

Reply:
> Tried that and almost pulled all my hair out. Can't remember why it didn't work. This grammar is mostly copied from the template parser.

or at least, I think that's where it's from... wherever I pulled that from doesn't even exist anymore